Ensure newaliases works out of the box

xavierba opened the pull request:
Hi Soji,

Here's a respin of my aliases branch with only the various .db files creation. I left out the sendmail/postfix alias merge and also the merge of /var/lib/sympa/aliases to /etc/sympa/aliases.

Regards,
Xavier

xavierba committed:
Ensure newaliases works out of the box

ikedas commented:
- `sympa_aliases.db` also have to be writable by sympa user (i.e. owned by sympa). Because Postfix's postalias try to derive permission of `sympa_aliases`.  Or, when list aliases are updated, it will fail with messsage
    ```
    postalias: fatal: open /path/to/sympa_aliases.db: Permission denied
    ```
  - Though it may be harmless, when problem above was solved, this warning is issued:
    ```
    postalias: warning: removing zero-length database file: /path/to/sympa_aliases.db
    ```
  - ~~sympa_aliases.db looks missing in %files.~~ My misleading.

xavierba commented:
I've updated the patch to force sympa_aliases.db to be sympa:root 664. Works with sendmail, can you please retest with postfix ? I also changed sympa_aliases to use the more standard 644 perms.

xavierba committed:
Fix create_config.

aliases files were not created when
/etc/sympa/data_structure.version was already created.
